src/lib.rs

```diff
@@ -167,47 +201,85 @@ impl Context {
         self.session = session_handle;
     }
 
-    pub fn create_primary_key(&mut self, auth_value: &[u8]) -> Result<ESYS_TR> {
-        if auth_value.len() > 64 {
+    // TODO: Fix when compacting the arguments into a struct
+    #[allow(clippy::too_many_arguments)]
+    pub fn create_primary_key(
+        &mut self,
+        sessions: Sessions,
+        primary_handle: ESYS_TR,
+        public: &TPM2B_PUBLIC,
+        auth_value: &[u8],
+        initial_data: &[u8],
+        outside_info: &[u8],
+        creation_pcrs: &[TPMS_PCR_SELECTION],
+    ) -> Result<ESYS_TR> {
+        if auth_value.len() > 64
+            || initial_data.len() > 256
+            || outside_info.len() > 64
+            || creation_pcrs.len() > 16
+        {
             return Err(Tss2ResponseCode::new(TPM2_RC_SIZE));
         }
-        let mut buffer = [0u8; 64];
-        buffer[..auth_value.len()].clone_from_slice(&auth_value[..auth_value.len()]);
+
+        let mut auth_value_buffer = [0u8; 64];
+        auth_value_buffer[..auth_value.len()].clone_from_slice(&auth_value[..auth_value.len()]);
+        let mut initial_data_buffer = [0u8; 256];
+        initial_data_buffer[..initial_data.len()]
+            .clone_from_slice(&initial_data[..initial_data.len()]);
         let sensitive_create = TPM2B_SENSITIVE_CREATE {
             size: std::mem::size_of::<TPMS_SENSITIVE_CREATE>()
                 .try_into()
                 .unwrap(),
             sensitive: TPMS_SENSITIVE_CREATE {
                 userAuth: TPM2B_DIGEST {
                     size: auth_value.len().try_into().unwrap(),
-                    buffer,
+                    buffer: auth_value_buffer,
+                },
+                data: TPM2B_SENSITIVE_DATA {
+                    size: initial_data.len().try_into().unwrap(),
```

Comment:
I guess this can never panic as `initial_data.len()` will never be bigger than 256?

Reply:
yes, should be alright